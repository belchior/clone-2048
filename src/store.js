import { createStore, applyMiddleware, compose } from 'redux';
import { loadState, saveState } from './localState';
import { reducer } from './reducers';


const saveStateMiddleware = ({ getState }) => next => (action) => {
  saveState(getState());
  return next(action);
};

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const store = createStore(
  reducer,
  loadState(),
  composeEnhancers(applyMiddleware(saveStateMiddleware))
);
