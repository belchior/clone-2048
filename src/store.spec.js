import { store, } from './store';

it('store should match propertys', () => {
  expect(store).toMatchSnapshot({
    dispatch: expect.any(Function),
    getState: expect.any(Function),
    replaceReducer: expect.any(Function),
    subscribe: expect.any(Function),
  });
});
