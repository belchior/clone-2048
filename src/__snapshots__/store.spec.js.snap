// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store should match propertys 1`] = `
Object {
  "dispatch": Any<Function>,
  "getState": Any<Function>,
  "replaceReducer": Any<Function>,
  "subscribe": Any<Function>,
  Symbol(observable): [Function],
}
`;
